models/repo_unit.go

```diff
@@ -16,11 +16,12 @@ import (
 
 // RepoUnit describes all units of a repository
 type RepoUnit struct {
-	ID          int64
-	RepoID      int64           `xorm:"INDEX(s)"`
-	Type        UnitType        `xorm:"INDEX(s)"`
-	Config      core.Conversion `xorm:"TEXT"`
-	CreatedUnix util.TimeStamp  `xorm:"INDEX CREATED"`
+	ID            int64
+	RepoID        int64    `xorm:"INDEX(s)"`
+	Type          UnitType `xorm:"INDEX(s)"`
+	AllowWriteAll bool
```

Comment:
We need a migration.

Reply:
I thought false is the default, then we would still need it?

Reply:
All database structure changes should have a migration. You can find many example on `models/migration`

Reply:
Thanks, I see, I somehow thought that there was some magic invovled which inserts missing columns automagically.

Reply:
There _is_ some magic involved, and you've probably got your column migrated automagically.

There are two stages to Gitea migration:

* The "manual steps" stage (from `models/migration`), that update the internal version and change data as needed.
* The "startup" stage, where any remaining database structure is updated.

The problem comes when you rely on the 2nd stage to do your migration. It will work the first time, but in the future (e.g. Gitea 12.2) some manual steps can be added in the first stage of the migration that will need the structure that is not there yet (as it is created in the 2nd stage). If some user attempts to migrate from 1.10 to 12.2, all the steps on stage 1 will be executed, and that particular step will fail because of the missing column.

That's why we don't rely on the 2nd stage and make all migration steps explicit, so any upcoming step can count on the previous steps to upgrade the database to the expected state.

I hope this makes sense!

Reply:
Thank you for a great explanation! Understood and I am convinced ;)